source/initial_composition/slab_model.cc

```diff
@@ -0,0 +1,123 @@
+      // In the input file, slab depths are to the top of the slab surface.
+      // The hyperbolic tangent function smooths the slabs in the depth direction.
+      if ( (compositional_index == slab_index) && (slab_depths < 1e10) && (slab_thickness < 1e10) &&
+           (depth >= slab_depths) && (depth <= slab_depths + slab_thickness) )
+        {
+          const double slab_center    = slab_depths + slab_thickness/2.;
+          const double half_thickness = slab_thickness/2;
+          slab_composition            = ( 1 - std::tanh( 10 * (std::abs (slab_center - depth) - half_thickness)/half_thickness ) )/2;
+        }
+
+      return slab_composition;
+    }
+
+
+    template <int dim>
+    void
+    SlabModel<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial composition model");
+      {
+        Utilities::AsciiDataBoundary<dim>::declare_parameters(prm,  "../../input_data/", "slab2_depth_thickness_2D.txt", "Slab model");
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    SlabModel<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial composition model");
+      {
+        slab_boundary.initialize_simulator (this->get_simulator());
+        slab_boundary.parse_parameters(prm, "Slab model");
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace InitialComposition
+  {
+    ASPECT_REGISTER_INITIAL_COMPOSITION_MODEL(SlabModel,
+                                              "slab model",
+                                              "An initial composition model computed to define slabs "
+                                              "using the slab2 model (Hayes et al., 2018) input in "
+                                              "an ascii format. The file describes the depths to the "
+                                              "top of the slabs and their thickness."
+                                              "The computed compositional value is 1 within the slabs "
+                                              "and zero elsewhere. In order to prevent sharp jumps, the "
+                                              "slabs are smoothened in the depth direction. "
+                                              "More details on the slab2 model can be found in "
+                                              "Hayes, G. P., Moore, G. L., Portner, D. E., Hearne, M., "
+                                              "Flamme, H., Furtney, M., & Smoczyk, G. M. (2018). Slab2, "
+                                              "a comprehensive subduction zone geometry model. Science, "
+                                              "362(6410), 58-61. The script to convert the slab2 model "
+                                              "into an aspect input data file is available.")
```

Comment:
```suggestion
                                              "into an aspect input data file is available "
                                              "in the directory data/initial_composition/slab/.")
```

Reply:
I think it should go into the data folder. We have some other python scripts already and they are mostly in the folders where they are needed. I see `contrib/python` more as a general folder for python scripts we do not know where else to put.